Makefile

```diff
@@ -24,37 +31,54 @@ GOOGLEAPIS_RPC_CCS = $(shell find $(GOOGLEAPIS_GENS_PATH)/google/rpc \
 
 GOOGLEAPIS_CCS = $(GOOGLEAPIS_API_CCS) $(GOOGLEAPIS_RPC_CCS) $(GOOGLEAPIS_TYPE_CCS)
 
-GOOGLEAPIS_ASSISTANT_CCS = ./googleapis/gens/google/assistant/embedded/v1alpha2/embedded_assistant.pb.cc \
-	./googleapis/gens/google/assistant/embedded/v1alpha2/embedded_assistant.grpc.pb.cc
+GOOGLEAPIS_ASSISTANT_CCS = $(GOOGLEAPIS_GENS_PATH)/$(GOOGLEAPIS_ASSISTANT_PATH)/embedded_assistant.pb.cc \
+                           $(GOOGLEAPIS_GENS_PATH)/$(GOOGLEAPIS_ASSISTANT_PATH)/embedded_assistant.grpc.pb.cc
 
 HOST_SYSTEM = $(shell uname | cut -f 1 -d_)
 SYSTEM ?= $(HOST_SYSTEM)
-CXX = g++
-CPPFLAGS += -I/usr/local/include -pthread -I$(GOOGLEAPIS_GENS_PATH) \
-	    -I$(GRPC_SRC_PATH) -I./src/
-CXXFLAGS += -std=c++11
+
+HAS_PKG_CONFIG ?= true
+ifeq ($(HAS_PKG_CONFIG),true)
+GRPC_GRPCPP_CFLAGS=`pkg-config --cflags grpc++ grpc`
+GRPC_GRPCPP_LDLAGS=`pkg-config --libs grpc++ grpc`
+ALSA_CFLAGS=`pkg-config --cflags alsa`
+ALSA_LDFLAGS=`pkg-config --libs alsa`
+else
+GRPC_GRPCPP_CFLAGS ?=
+GRPC_GRPCPP_LDLAGS ?=
+ALSA_CFLAGS ?=
+ALSA_LDFLAGS ?=
+endif
+
+CPPFLAGS += -I$(GOOGLEAPIS_GENS_PATH) \
+            -I$(GRPC_SRC_PATH) \
+            -I./src/
+
+CXXFLAGS += -std=c++11 $(GRPC_GRPCPP_CFLAGS)
+
 # grpc_cronet is for JSON functions in gRPC library.
 ifeq ($(SYSTEM),Darwin)
-LDFLAGS += -L/usr/local/lib `pkg-config --libs grpc++ grpc`       \
-           -lgrpc++_reflection -lgrpc_cronet                      \
+LDFLAGS += $(GRPC_GRPCPP_LDLAGS) \
+           -lgrpc++_reflection -lgrpc_cronet \
            -lprotobuf -lpthread -ldl -lcurl
 else
-LDFLAGS += -L/usr/local/lib `pkg-config --libs grpc++ grpc`       \
-           -lgrpc_cronet -Wl,--no-as-needed -lgrpc++_reflection   \
+LDFLAGS += $(GRPC_GRPCPP_LDLAGS) \
+           -lgrpc_cronet -Wl,--no-as-needed -lgrpc++_reflection \
            -Wl,--as-needed -lprotobuf -lpthread -ldl -lcurl
 endif
 
 AUDIO_SRCS =
 ifeq ($(SYSTEM),Linux)
 AUDIO_SRCS += src/audio_input_alsa.cc src/audio_output_alsa.cc
-LDFLAGS += `pkg-config --libs alsa`
+CXXFLAGS += $(ALSA_CFLAGS)
+LDFLAGS += $(ALSA_LDFLAGS)
 endif
 
 .PHONY: all
 all: run_assistant
 
 googleapis.ar: $(GOOGLEAPIS_CCS:.cc=.o)
-	ar r $@ $?
+	$(AR) r $@ $?
```

Comment:
Where is `$(AR)` defined?

Reply:
`$(AR)` is a pre-defined variable in GNU make, like `$(CXX)`, see [here](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html). By hard coding `ar` like this, there is no chance to properly cross-compile the SDK by passing its value from outside the Makefile.